### PR TITLE
HOMME: 3D semi-Lagrangian transport

### PR DESCRIPTION
**Pull request by ambrad**
Combines
- 3D SL transport
- SL memory footprint reduction

Introduces dt_remap_factor, dt_tracer_factor. Permits SL transport times steps to be longer than vertical remap time steps.

[BFB] for non-SL tests
[non-BFB] for SL: 4 new tests, removed 1 old test, will diff for 1 existing test.

### CODE REVIEW

**Comment by ambrad** [comment]
[Confluence page](https://acme-climate.atlassian.net/wiki/spaces/COM/pages/1041630611/SL+tracer+transport+time+step+independent+of+dynamics+vertical+remap+time+step) with climotology results.

**Comment by ambrad** [comment]
e3sm_developer gives:
```
SMS.r05_r05.I1850CLM45CN.sandiatoss3_intel FAIL
The following tests FAILED:
    1 - verifyBaselineResults (Failed)
   21 - thetah-sl-test11conv-r1t2-cdr20 (Failed)
   22 - thetah-sl-test11conv-r0t1-cdr30-rrm (Failed)
   23 - thetah-sl-dcmip16_test1pg2 (Failed)
```
The `eam_theta` `thetahy_sl` test will diff. There is a new e3sm_developer test `theta_sl_hy`. The SMS test failure above above seems unrelated to this PR.

File: components/homme/src/share/control_mod.F90 [hunk 1]
```diff
@@ -230,4 +246,365 @@ module control_mod
...
+    if (present(silent)) silent_in = silent
+
+    status = -1 ! error value for early returns on error
+
+    !! Process dtime, tstep, nsplit.
+    dt_max_factor = max(dt_remap_factor, dt_tracer_factor)
+
+    ! Every 'if' has an 'else', so every case is covered.
+    if (nsplit > 0) nstep_factor = dt_max_factor*nsplit
+    if (dtime > 0) then
```
Comment (oksanaguba) [inline components/homme/src/share/control_mod.F90]: I could not trace this: why would dtime be < 0?

**Comment by oksanaguba** [comment]
It is probably nothing, I tried to merge this branch to master and got
```
fatal: reference is not a tree: 31ff567f422c2881aae5d5c0db8d4563266ddaa2
Unable to checkout '31ff567f422c2881aae5d5c0db8d4563266ddaa2' in submodule path 'externals/kokkos'```

File: components/cam/cime_config/testdefs/testmods_dirs/cam/thetahy_sl_pg2/user_nl_cam [hunk 2]
```diff
@@ -0,0 +1,17 @@
+theta_hydrostatic_mode = .true.
+se_fv_phys_remap_alg = 1
+tstep_type = 5
+theta_advect_form = 1
+hypervis_subcycle_tom = 1
+hypervis_subcycle = 1
+transport_alg = 12
+semi_lagrange_cdr_alg = 20
```
Comment (oksanaguba) [inline components/cam/cime_config/testdefs/testmods_dirs/cam/thetahy_sl_pg2/user_nl_cam]: why not 21? is 21 the most advanced now?

Comment (ambrad) [reply components/cam/cime_config/testdefs/testmods_dirs/cam/thetahy_sl_pg2/user_nl_cam]: Good question. 21 is a bit of a pet project for me. I want it available for future work. But for now, I want to stick with the tried-and-true 20.

**Comment by ambrad** [comment]
Re: merge issue, I just tried the same and had no issue. That ref is for Kokkos, so it's definitely trying to do something with the kokkos submodule. You might run `git submodule sync`, `git submodule update` in your master and try again.

File: components/homme/src/share/control_mod.F90 (hunk 1 above)
Comment (ambrad) [reply components/homme/src/share/control_mod.F90]: It is meaningless in standalone HOMME. For testing and possible future use, I want dtime=-1 to signify it should be deduced (e.g., if se_nsplit and time are specified, and dtime=-1).

**Comment by oksanaguba** [comment]
I ran submodules stuff before merging.

**Comment by oksanaguba** [comment]
This user_nl_cam did not work out:
```
dt_remap_factor=2
tstep=75
dt_tracer_factor=6
```
error was
```
   0:  ERROR:
   0:  homme:namelist_mod:: namelist read returns an end of file or end of record cond
   0:  ition
   0: Image              PC                Routine            Line        Source
   0: e3sm.exe           000000000377D2FB  Unknown               Unknown  Unknown
   0: e3sm.exe           0000000002445952  shr_abort_mod_mp_         114  shr_abort_mod.F90
   0: e3sm.exe           000000000170025E  namelist_mod_mp_r         423  namelist_mod.F90
```
Instead of using tstep, i used se_nsplit=2 . However, we may want to use tstep in CAM.

**Comment by ambrad** [comment]
Thanks. I'll modify namelist_mod to permit tstep to be provided when CAM is defined.

**Comment by ambrad** [comment]
@mt5555, @oksanaguba, is it OK to expose `tstep` -- i.e., exactly that word -- in atm_in? Or should I call it se_tstep or something else?

**Comment by ambrad** [comment]
The last commit brings in se_tstep. I reran e3sm_developer and get the same as in my first run of it.

**Comment by oksanaguba** [comment]
Tried this branch merged to master again with
```
dt_remap_factor=2
se_tstep=75
dt_tracer_factor=6 
```
and all as expected:
```
dt_remap: (0=disabled)      150.00
dt_tracer (SE), per RK stage:    450.00   225.00
dt_dyn:                      75.00
```

**Pull request merged**